pkgs/state-proving-manager/src/manager.zig

```diff
@@ -1,15 +1,141 @@
-const types = @import("types")
-const state_transition = @import("state-transition")
+const std = @import("std");
+const types = @import("types");
+const ssz = @import("ssz");
+const state_transition = @import("state-transition");
+const allocator = std.heap.page_allocator;
 
+pub const ZKVMContext = struct {
+    program_path: []const u8,
+    output_dir: []const u8,
+    backend: []const u8,
+};
 
-const StateTransitionOpts = struct{
-    zk_vm: ZKVM_TYPES,
-}
+pub const zkvm_configs: []ZKVMContext = .{
+    .{ .program_path = "dist/zeam-stf-powdr.asm", .output_dir = "out", .backend = "plonky3" },
+};
+
+const StateTransitionOpts = struct {
+    zk_vm: ZKVMContext,
+};
+
+pub fn execute_transition(state: types.BeamState, block: types.SignedBeamBlock, opts: StateTransitionOpts) !types.BeamSTFProof {
+    var publics = std.ArrayList(u8).init(allocator);
+
+    var serialized = try std.ArrayList(u8).init(allocator);
+    defer serialized.deinit();
+    try ssz.serialize(types.SignedBeamBlock, block, serialized);
+
+    var input = try std.fmt.allocPrint(allocator, "{any}", .{serialized});
+
+    // Run the executable without proving, in order to recover the publics
+    var dryrun_process = std.process.Child.init(&[_][]const u8{
```

Comment:
ahh nice so we automating your poc demo here!
zkVMs would eventually be providing this programmatic api?

Reply:
yes, it's currently possible to do that with rust libs - which I will work on integrating in a subsequent PR - right now I just want to do the whole end-to-end thing and calling external programs will do.